File: test72_out/pollen.event/Dispatcher/Dispatcher.c

/*
 * ======== string literals ========
 */

static const char pollen_event_Dispatcher__s__fileName[] = "Dispatcher.p";
static const char pollen_event_Dispatcher__s__1[] = "Dispatcher";
static const char pollen_event_Dispatcher__s__0[] = "Event registered. Capacity=";

/*
 * ======== function definitions ========
 */

void pollen_event_Dispatcher_postFromInterrupt__E( struct pollen_event_Event*  e ) {
    pollen_event_EventQueue_EventQueue_add__E(&(pollen_event_Dispatcher_events__V), (e));
}

void pollen_event_Dispatcher_dispatch__I() {
    pollen_event_Event_ e;
    e = null;
    (e) = pollen_event_EventQueue_EventQueue_remove__E(&(pollen_event_Dispatcher_events__V));
    while ((e) != null) {
        printf("Dispatching..");

        virtual_mcu_GlobalInterrupts_enable__E();
        pollen_event_Event_Event_fire__E(e);
        (e) = pollen_event_EventQueue_EventQueue_remove__E(&(pollen_event_Dispatcher_events__V));
    }
}

void pollen_event_Dispatcher_post__E( struct pollen_event_Event*  e ) {
    printf("Event posted..");

    virtual_mcu_GlobalInterrupts_disable__E();
    pollen_event_EventQueue_EventQueue_add__E(&(pollen_event_Dispatcher_events__V), (e));
    virtual_mcu_GlobalInterrupts_enable__E();
}

void pollen_event_Dispatcher_run__E() {
    virtual_mcu_GlobalInterrupts_enable__E();
    while (true) {
        virtual_mcu_GlobalInterrupts_disable__E();
        pollen_event_Dispatcher_dispatch__I();
        virtual_mcu_GlobalInterrupts_enable__E();
        break;
    }
}

void pollen_event_Dispatcher_targetInit__I() {
}

File: test72_out/pollen.event/Dispatcher/Dispatcher.js

/*
 * ======== MODULE pollen.event.Dispatcher ========
 */

var $$c = function() {
    const Dispatcher = {};
    Dispatcher.Dispatcher = Dispatcher
    Dispatcher.$name = 'pollen.event.Dispatcher';
    Dispatcher.pollen$used = 0;
    Dispatcher.Event = $units['pollen.event.Event'];
    Dispatcher.EventQueue = $units['pollen.event.EventQueue'];
    Dispatcher.targetInit = new $$Ref('pollen_event_Dispatcher_targetInit__I');
    Dispatcher.post = new $$Ref('pollen_event_Dispatcher_post__E');
    Dispatcher.postFromInterrupt = new $$Ref('pollen_event_Dispatcher_postFromInterrupt__E');
    Dispatcher.run = new $$Ref('pollen_event_Dispatcher_run__E');
    Dispatcher.dispatch = new $$Ref('pollen_event_Dispatcher_dispatch__I');
    Dispatcher.$$hostInit = function() {
        var $$text = '';
        Dispatcher.events = new $units['pollen.event.EventQueue'].EventQueue('pollen_event_Dispatcher_events__V'); Dispatcher.events.new_host((Dispatcher.capacity));
        return $$text;
    }
    Dispatcher.registerEvent = function() {
        var $$text = '';
        Dispatcher.events.setCapacityOnHost(++(Dispatcher.capacity));
        $$printf("Event registered. Capacity=", (Dispatcher.capacity));
        return $$text;
    }
    Dispatcher.pollen__uses__ = function() {
        $$bind($units['pollen.event.Event'], 'pollen$used', true);
        $$bind($units['pollen.event.EventQueue'], 'pollen$used', true);
    }
    Dispatcher.$$privateInit = function() {
        Dispatcher.capacity = 5;
        Dispatcher.GI = 0;
    }
    return Dispatcher;
}
var $$u = $$c();
$units['pollen.event.Dispatcher'] = $$u;
$units.push($$u);

